### ISSUE TITLE
feat(jsonrpc): Implement backwards compatible structured errors

### ISSUE THREAD

**Issue by frol** [near/nearcore#3517]
* [x] ~Rebase #3204 to the latest master~ *(moved into a separate issue #3518)*
* [x] Adapt JSON RPC responses so they return text old errors based on the new view client detailed errors (the structure and the messages must be IDENTICAL to the previous response messages); to do that we should match the structured error response received from the view client and craft the text of the old error message

This issue should be considered to be resolved once we can merge the proposed changes to the view-client (intorducing detailed programmatic errors instead of string errors), so any errornous response MUST remain the same:

```
$ http post https://rpc.testnet.near.org jsonrpc=2.0 id=dontcare method=block 'params:=[0]'

{
    "error": {
        "code": -32000,
        "data": "DB Not Found Error: BLOCK HEIGHT: 0 \n Cause: Unknown",
        "message": "Server error"
    },
    "id": "dontcare",
    "jsonrpc": "2.0"
}
```

We are going to expose those structured errors via RPC side-by-side with the current text errors as part of #2976. We do it in two steps to split the effort and take some time to bikeshed the structure we want to have for the RPC interfaces.

# Implementation Notes

I propose to implement `.to_old_error_message() -> String` method for every error type we introduced for the view-client, so ultimately, on the `jsonrpc` side we can just call `.to_old_error_message()` on the `ViewClientError` and it would just recursively construct the error message for us.

**Comment by bowenwang1996** [comment]
>  Rebase #3204 to the latest master

@frol I think this is not a trivial task and it may be worth creating a separate issue for it. I can help with the task.

**Comment by frol** [comment]
@bowenwang1996 I see. Here you go https://github.com/near/nearcore/issues/3518 :smile:

**Comment by frol** [comment]
This is resolved in a set or PRs: #3805, #3861, #3944, #4079, #4119, #4180, #4200, and #4213.

The next step is #2976
